utilities/new_client.py

```diff
@@ -0,0 +1,297 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Usage: python new_client.py -v <api version> -s <service name> -c <path to config>
+
+import argparse
+import copy
+import os
+import re
+import sys
+import subprocess
+from jinja2 import Environment, FileSystemLoader
+from lxml import etree
+from releasetool.commands.start import java as releasetool
+from typing import List
+
+class Context:
+    service: str = None
+    api_version: str = None
+    artman_config: str = None
+    google_cloud_artifact: str = None
+    google_cloud_version: releasetool.Version = None
+    grpc_artifact: str = None
+    grpc_version: releasetool.Version = None
+    proto_artifact: str = None
+    proto_version: releasetool.Version = None
+    root_directory: str = None
+    description: str = "FIXME"
+    name: str = "FIXME"
+    versions: List[str] = None
+
+    def __init__(self, service: str, api_version: str, artman_config: str):
+        self.service = service
+        self.api_version = api_version
+        self.artman_config = artman_config
+        self.google_cloud_artifact = f'google-cloud-{service}'
+        self.grpc_artifact = f'grpc-google-cloud-{service}-{api_version}'
+        self.proto_artifact = f'proto-google-cloud-{service}-{api_version}'
+        self.root_directory = os.path.realpath(os.path.join(os.path.dirname(os.path.realpath(__file__)), '..'))
```

Comment:
If you make this a pathlib.Path, you can just use `ctx.root / "whatever"` instead of `ctx.path("whatever")`.